// @mui/x-date-pickers-pro/esm/managers/useDateRangeManager.js
'use client';

import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { usePickerTranslations } from '@mui/x-date-pickers/hooks';
import { useApplyDefaultValuesToDateValidationProps, useUtils } from '@mui/x-date-pickers/internals';
import { getRangeFieldValueManager, rangeValueManager } from "../internals/utils/valueManagers.js";
import { validateDateRange } from "../validation/index.js";
import { formatRange } from "../internals/utils/date-utils.js";
export function useDateRangeManager(parameters = {}) {
  const {
    enableAccessibleFieldDOMStructure = true,
    dateSeparator
  } = parameters;
  return React.useMemo(() => ({
    valueType: 'date',
    validator: validateDateRange,
    internal_valueManager: rangeValueManager,
    internal_fieldValueManager: getRangeFieldValueManager({
      dateSeparator
    }),
    internal_enableAccessibleFieldDOMStructure: enableAccessibleFieldDOMStructure,
    internal_useApplyDefaultValuesToFieldInternalProps: useApplyDefaultValuesToDateRangeFieldInternalProps,
    internal_useOpenPickerButtonAriaLabel: useOpenPickerButtonAriaLabel
  }), [enableAccessibleFieldDOMStructure, dateSeparator]);
}
function useOpenPickerButtonAriaLabel(value) {
  const utils = useUtils();
  const translations = usePickerTranslations();
  return React.useMemo(() => {
    return translations.openRangePickerDialogue(formatRange(utils, value, 'fullDate'));
  }, [value, translations, utils]);
}
function useApplyDefaultValuesToDateRangeFieldInternalProps(internalProps) {
  const utils = useUtils();
  const validationProps = useApplyDefaultValuesToDateValidationProps(internalProps);
  return React.useMemo(() => _extends({}, internalProps, validationProps, {
    format: internalProps.format ?? utils.formats.keyboardDate
  }), [internalProps, validationProps, utils]);
}
