// @mui/x-date-pickers-pro/esm/managers/useTimeRangeManager.js
'use client';

import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from 'react';
import { usePickerTranslations } from '@mui/x-date-pickers/hooks';
import { useApplyDefaultValuesToTimeValidationProps, useUtils } from '@mui/x-date-pickers/internals';
import { getRangeFieldValueManager, rangeValueManager } from "../internals/utils/valueManagers.js";
import { validateTimeRange } from "../validation/index.js";
import { formatRange } from "../internals/utils/date-utils.js";
export function useTimeRangeManager(parameters = {}) {
  const {
    enableAccessibleFieldDOMStructure = true,
    dateSeparator,
    ampm
  } = parameters;
  return React.useMemo(() => ({
    valueType: 'time',
    validator: validateTimeRange,
    internal_valueManager: rangeValueManager,
    internal_fieldValueManager: getRangeFieldValueManager({
      dateSeparator
    }),
    internal_enableAccessibleFieldDOMStructure: enableAccessibleFieldDOMStructure,
    internal_useApplyDefaultValuesToFieldInternalProps: useApplyDefaultValuesToTimeRangeFieldInternalProps,
    internal_useOpenPickerButtonAriaLabel: createUseOpenPickerButtonAriaLabel(ampm)
  }), [enableAccessibleFieldDOMStructure, dateSeparator, ampm]);
}
function createUseOpenPickerButtonAriaLabel(ampm) {
  return function useOpenPickerButtonAriaLabel(value) {
    const utils = useUtils();
    const translations = usePickerTranslations();
    return React.useMemo(() => {
      const formatKey = ampm ?? utils.is12HourCycleInCurrentLocale() ? 'fullTime12h' : 'fullTime24h';
      return translations.openRangePickerDialogue(formatRange(utils, value, formatKey));
    }, [value, translations, utils]);
  };
}
function useApplyDefaultValuesToTimeRangeFieldInternalProps(internalProps) {
  const utils = useUtils();
  const validationProps = useApplyDefaultValuesToTimeValidationProps(internalProps);
  const ampm = React.useMemo(() => internalProps.ampm ?? utils.is12HourCycleInCurrentLocale(), [internalProps.ampm, utils]);
  return React.useMemo(() => _extends({}, internalProps, validationProps, {
    format: internalProps.format ?? (ampm ? utils.formats.fullTime12h : utils.formats.fullTime24h)
  }), [internalProps, validationProps, ampm, utils]);
}
